Source/SwiftLintBuiltInRules/Rules/Style/MultilineParametersRuleExamples.swift

```diff
@@ -336,5 +338,10 @@ internal struct MultilineParametersRuleExamples {
                 configuration: ["allows_single_line": false]),
         Example("func ↓foo(param1: Int, param2: Bool, param3: [String]) { }",
                 configuration: ["allows_single_line": false]),
+        Example("""
+        func ↓foo(param1: Int,
+                  param2: Bool, param3: [String]) { }
```

Comment:
I'm trying to make sense of this new option. Why would it be helpful to allow this style given `max_number_of_params = 3`?

Reply:
Hmm. Maybe we should only allow 3 params etc, if they are on the same line?
Would that make more sense?

Reply:
We could in this case do the other way around?

So if it's set to 3, and the there is a newline then it will show a warning to put on a single line?
But maybe it would be better in that case to show a warning that all params should be on a newline instead?

Reply:
The default style the rule enforces is that all parameters are on the same line or every parameter is on a separate line.

Optionally, with `allows_single_line = false`, all parameters need to be placed on their own line.

With the new option, you want to allow a specified number of parameters to be placed on a single line. Once the threshold is exceeded, all parameters are required to be placed on a separate line again. That's basically `allows_single_line = true` up to a number of parameters.

I think that enforcing a single line is not practicable, because parameters and their types could consume a lot of space and so placing them on separate line, even without the rule enforcing it, should be fine.

Now, how does `max_number_of_parameters` (maybe we need a better name) play nicely with `allows_single_line = false`? It's a contradiction. On the other hand, `allows_single_line = true` is redundant with `max_number_of_parameters` being configured with a reasonable number. We somehow need to align both, deprecate or reuse `allows_single_line`.

Reply:
Okay so I rearranged some code. 

I still think that `allows_single_line` and `max_number_of_parameters` can live next to each other. 
It's just, when `allows_single_line` is `false` then `max_number_of_parameters` is ignored. I think that is fine. 

I added a test case, where allow single line is false but there is fewer params than  `max_number_of_parameters` define.
Also I added one where there is 3 params on each line. Also within the allow number in `max_number_of_parameters`.

Reply:
@SimplyDanny do you have time to review again? :)

Reply:
So the intention is that single lines can be okay as long as they fit and a certain number of parameters is not exceeded. That seems to be fine.

What about the option's name? Would `max_number_of_single_line_parameters` be clearer in its meaning?

Another improvement would be a warning if `allows_single_line = false` and `max_number_of_parameters > 1`. That can be added later, though, as I need to think first about how to support that in a generic way.

Reply:
> Another improvement would be a warning if `allows_single_line = false` and `max_number_of_parameters > 1`. That can be added later, though, as I need to think first about how to support that in a generic way.

Perhaps something like #5824?

Reply:
> What about the option's name? Would max_number_of_single_line_parameters be clearer in its meaning?

This is now added. 

> Another improvement would be a warning if allows_single_line = false and max_number_of_parameters > 1. That can be added later, though, as I need to think first about how to support that in a generic way.

Should like a good idea to have something where it's possible to give a warnings on clashing configurations. 
I will wait till it's merged, so we can add it right await

Reply:
https://github.com/realm/SwiftLint/pull/5824 has just been merged. Let me know if it's sufficient.

Reply:
I've added configuration. Please revisit when you have time.